boards/arm/segger_trb_stm32f407/doc/index.rst

```diff
@@ -0,0 +1,138 @@
+.. _segger_trb_stm32f407:
+
+SEGGER-TRM-STM32F407
```

Comment:
The official name seems to be "[Cortex-M Trace Reference Board V1.2](https://shop.segger.com/Cortex_M_Trace_Reference_Board_p/6.68.18.htm)" and should be used here,  abbreviation could be introduced later in overview chapter.

Reply:
Sadly it is a terrible long name. But as "title" it seems to be standard to use the "real" name as far as I can see, so i'll update it. 
Just to clarify the idea between the `segger_trb_stm32f407` name, it includes the producer, and they have 2 others versions, one with a imx rt1050 and one with a stm32h7. so there could be consistent dir/board naming like;
- segger_trb_stm32f407
- segger_trb_imxrt1050
- segger_trb_stm32h743